src/app/DIAmeterApplication.cpp

```diff
@@ -73,241 +80,871 @@ int DIAmeterApplication::main(const vector<string>& input_files, const string in
+
+				map<int, boost::tuple<double*, double*, int>> ms2scan_mz_intensity_map;
+				buildSpectraIndexFromIsoWindowNew(&spec_charge_chunk, &ms2scan_mz_intensity_map);
+
+
+				for (vector<SpectrumCollection::SpecCharge>::const_iterator sc = spec_charge_chunk.begin();sc < spec_charge_chunk.begin() + (spec_charge_chunk.size()); sc++)
+				{
+					Spectrum* spectrum = sc->spectrum;
+					double precursor_mz = spectrum->PrecursorMZ();
+					int scan_num = spectrum->SpectrumNumber();
+					int ms1_scan_num = spectrum->MS1SpectrumNum();
+					int charge = sc->charge;
+
+					// The active peptide queue holds the candidate peptides for spectrum.
+					// Calculate and set the window, depending on the window type.
+					vector<double>* min_mass = new vector<double>();
+					vector<double>* max_mass = new vector<double>();
+					vector<bool>* candidatePeptideStatus = new vector<bool>();
+					double min_range, max_range;
+
+					// TideSearchApplication::computeWindow(*sc, string_to_window_type(Params::GetString("precursor-window-type")), Params::GetDouble("precursor-window"), Params::GetInt("max-precursor-charge"), &negative_isotope_errors, min_mass, max_mass, &min_range, &max_range);
+					// carp(CARP_DETAILED_DEBUG, "==============MS1Scan:%d \t MS2Scan:%d \t precursor_mz:%f \t charge:%d", ms1_scan_num, scan_num, precursor_mz, charge);
+					computeWindowDIA(*sc, &negative_isotope_errors, min_mass, max_mass, &min_range, &max_range);
+
+
+					// Normalize the observed spectrum and compute the cache of frequently-needed
+					// values for taking dot products with theoretical spectra.
+					// TODO: Note that here each specturm might be preprocessed multiple times, one for each charge, potentially can be improved!
+					observed.PreprocessSpectrum(*spectrum, charge, &num_range_skipped, &num_precursors_skipped, &num_isotopes_skipped, &num_retained);
+					int nCandPeptide = active_peptide_queue->SetActiveRange(min_mass, max_mass, min_range, max_range, candidatePeptideStatus);
+					int candidatePeptideStatusSize = candidatePeptideStatus->size();
+					// carp(CARP_DETAILED_DEBUG, "nCandPeptide:%d \t candidatePeptideStatusSize:%d \t mass range:[%f,%f]", nCandPeptide, candidatePeptideStatusSize, min_range, max_range);
+					if (nCandPeptide == 0) { continue; }
+
+					// carp(CARP_DETAILED_DEBUG, "==============min_mass:%s \t max_mass:%s", StringUtils::Join(*min_mass, ',').c_str(), StringUtils::Join(*max_mass, ',').c_str()  );
+					// carp(CARP_DETAILED_DEBUG, "==============nCandPeptide:%d \t candidatePeptideStatusSize:%d \t mass_range:[%f,%f]", nCandPeptide, candidatePeptideStatusSize, min_range, max_range);
+
+
+					TideMatchSet::Arr2 match_arr2(candidatePeptideStatusSize); // Scored peptides will go here.
+					// Programs for taking the dot-product with the observed spectrum are laid
+					// out in memory managed by the active_peptide_queue, one Januaryprogram for each
```

Comment:
I may just be ignorant but is "Januaryprogram" the right term? I have never heard of it